google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/models.scala

```diff
@@ -6,9 +6,33 @@ package akka.stream.alpakka.googlecloud.pubsub
 
 import java.time.Instant
 
+import akka.actor.ActorSystem
+import akka.annotation.InternalApi
+import akka.http.scaladsl.{Http, HttpExt}
+import akka.stream.alpakka.googlecloud.pubsub.impl.{GoogleSession, GoogleTokenApi}
+
 import scala.collection.immutable
 import scala.collection.JavaConverters._
 
+class PubSubConfig(val projectId: String,
+                   val apiKey: String,
+                   clientEmail: String,
+                   privateKey: String,
+                   http: => HttpExt) {
+  def this(projectId: String, apiKey: String, clientEmail: String, privateKey: String)(
```

Comment:
I think this is not needed. The Http extension access can be moved to the apply method.